unit_tests/log/test_main_data_page_fetcher.cpp

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2008 Search Solution Corporation
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#define CATCH_CONFIG_MAIN
+
+#include "catch2/catch.hpp"
+#include "async_page_fetcher.hpp"
+#include "dbtype_def.h"
+#include "log_reader.hpp"
+#include "page_buffer.h"
+
+#include <mutex>
+#include <unordered_map>
+#include <vector>
+
+struct page_id_requirement
+{
+  bool require_data_page_valid;
+  bool is_page_received;
+};
+
+struct data_page_fetcher_test_data
+{
+  std::unordered_map<int64_t, page_id_requirement> page_ids_requested;
+  std::mutex map_mutex;
+} g_data_page_fetcher_test_data;
+
+int64_t get_key (VPID page_id)
+{
+  int64_t key = page_id.pageid;
+  key = (key << (sizeof (short) * 8)) | page_id.volid;
+  return key;
+}
+
+class test_env
+{
+  public:
+    test_env (bool require_data_page_valid, std::vector<VPID> vpids);
+    ~test_env ();
+
+    void run_test ();
+
+  private:
+    void on_receive_data_page (VPID page_id, const FILEIO_PAGE *data_page, int error_code);
+
+  private:
+    std::vector<VPID> m_vpids;
+    cublog::async_page_fetcher m_async_page_fetcher;
+};
+
+void do_test (test_env &env)
+{
+  env.run_test ();
+}
+
+TEST_CASE ("Test with a valid log page returned", "")
```

Comment:
Test case covers data pages, not log pages, and not only valid cases.
```suggestion
TEST_CASE ("Test with a valid log page returned", "")
```